Makefile

```diff
@@ -157,9 +165,47 @@ cmctl: ## Downloads and installs cmctl, the CLI for cert-manager, to your local
 	rm -rf $$TMP_DIR ;\
 	}
 
+.PHONY: cert-manager
+cert-manager: cmctl ## Ensures cert-manager is deployed in the current Kubernetes cluster and verifies API access with kubectl wait
+	@if ! kubectl get pods -l app=cert-manager --all-namespaces | grep -q 'Running'; then \
+		echo "Deploying cert-manager now..."; \
+		kubectl apply --validate=false -f https://github.com/jetstack/cert-manager/releases/download/${CERTMANAGER_VERSION}/cert-manager.yaml || exit 1; \
+		echo "Waiting until cert-manager is running."; \
+	  sleep 5; \
+	else \
+		echo "cert-manager is already deployed."; \
+	fi; \
+	$(CMCTL) check api --wait=5m; \
+
+.PHONY: cert-manager-cleanup
+cert-manager-cleanup: ## Ensures cert-manager is deleted in the current Kubernetes cluster and ensures all related resources are removed
+	kubectl delete issuers.cert-manager.io --all --all-namespaces || true
+	kubectl delete clusterissuers.cert-manager.io --all --all-namespaces || true
+	kubectl delete certificates.cert-manager.io --all --all-namespaces || true
+	kubectl delete certificaterequests.cert-manager.io --all --all-namespaces || true
+	kubectl delete orders.acme.cert-manager.io --all --all-namespaces || true
+	kubectl delete challenges.acme.cert-manager.io --all --all-namespaces || true
+	kubectl delete -f https://github.com/jetstack/cert-manager/releases/download/${CERTMANAGER_VERSION}/cert-manager.yaml || true
```

Comment:
This should remove all the resources, right? why do we need to replicate the same manipulations in the other steps?

Reply:
It won't remove the certmanager content (Issuers,ClusterIssuers,Certificates,CertificateRequests,Orders,Challenges) the user created. The certmanager docs mention this at https://cert-manager.io/docs/installation/kubectl/#uninstalling.

Reply:
- This command is valuable for ensuring that a Kubernetes cluster is sufficiently cleaned to deploy a new certmanager. If any outdated objects such as "Issuers, ClusterIssuers, Certificates, CertificateRequests, Orders, Challenges" persist in the cluster, they could complicate the deployment of the new certmanager.
- Although the cleanup process may seem extensive, it's necessary in certain situations to prevent deployment issues.